backend/schema/validate.go

```diff
@@ -38,15 +38,21 @@ var (
 //
 // This is useful for testing.
 func MustValidate(schema *Schema) *Schema {
-	clone, err := Validate(schema)
+	clone, err := ValidateWholeSchema(schema)
 	if err != nil {
 		panic(err)
 	}
 	return clone
 }
 
-// Validate clones, normalises and semantically valies a schema.
-func Validate(schema *Schema) (*Schema, error) {
+// ValidateWholeSchema clones, normalises and semantically validates a schema.
+func ValidateWholeSchema(schema *Schema) (*Schema, error) {
+	return ValidateModuleInSchema(schema, nil)
+}
+
+// ValidateModuleInSchema clones and normalises a schema and semantically validates a single module within it.
+// If no module is provided, all modules in the schema are validated.
+func ValidateModuleInSchema(schema *Schema, m *Module) (*Schema, error) {
```

Comment:
I'm not 100% consistent with this myself, but I do try to use `Optional` in places where it conveys semantic meaning like it does here.

Reply:
good to know, ty!